packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -687,22 +591,27 @@ export class SDKClient {
           .setContents(hexedCallData.substring(this.fileAppendChunkSize, hexedCallData.length))
           .setChunkSize(this.fileAppendChunkSize)
           .setMaxChunks(this.maxChunks);
-        const fileAppendTxResponse = await fileAppendTx.execute(client);
-
-        // get transaction fee and add expense to limiter
-        const appendFileRecord = await fileAppendTxResponse.getRecord(this.clientMain);
-        transactionFee = appendFileRecord.transactionFee;
-        this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
-
-        this.captureMetrics(
-          SDKClient.transactionMode,
-          fileAppendTx.constructor.name,
-          Status.Success,
-          await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
-          0,
-          callerName,
-          interactingEntity,
-        );
+        const fileAppendTxResponses = await fileAppendTx.executeAll(client);
+
+        let totalTinybarsCost = 0;
+        for (let fileAppendTxResponse of fileAppendTxResponses) {
+          // get transaction fee and add expense to limiter
+          const appendFileRecord = await fileAppendTxResponse.getRecord(this.clientMain);
+          totalTinybarsCost += appendFileRecord.transactionFee.toTinybars().toNumber();
+        }
+
+        if (totalTinybarsCost > 0) {
+          this.hbarLimiter.addExpense(totalTinybarsCost, currentDateNow);
+          this.captureMetrics(
```

Comment:
Q: Doesn't this throw metrics of transaction count off? 
The below will capture 1 FileAppend count metric but indeed will capture the hbar expense of the collection File appends.

Reply:
I changed it so it executes `addExpense` and `captureMetrics` for every single `fileAppendTransaction`